src/common/tracer.cc

```diff
@@ -55,6 +65,50 @@ bool Tracer::is_enabled() const {
   return g_ceph_context->_conf->jaeger_tracing_enable;
 }
 
+void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f) {
+  if (span_ctx.IsValid()) {
+    using namespace opentelemetry;
+    using namespace trace;
+    const size_t ctx_size = TraceId::kSize + SpanId::kSize + 1;
+    char trace_identity[ctx_size];
+    uint8_t* trace_identity_ptr = (uint8_t*)trace_identity;
```

Comment:
why do you need this conversion? why not use char?

Reply:
`CopyBytesTo` method of TraceId/SpanId class gets unsigned char. so we can't use char.

Reply:
i guess that if we don't use a buffer and encode/decode explicitly the different values, then this does not matter